dsl_parser/functions.py

```diff
@@ -588,22 +597,88 @@ def __init__(self, args, **kwargs):
         super(GetSecret, self).__init__(args, **kwargs)
 
     def parse_args(self, args):
-        if not isinstance(args, basestring) and not is_function(args):
+        if not isinstance(args, string_types) and not is_function(args):
             raise ValueError(
                 "`get_secret` function argument should be a string\\dict "
                 "(a function). Instead it is a {0} with the "
                 "value: {1}.".format(type(args), args))
         self.secret_id = args
 
+    def evaluate(self, handler):
+        return handler.get_secret(self.secret_id)
+
+
+@register(name='concat', func_eval_type=HYBRID_FUNC)
+class Concat(Function):
+
+    def __init__(self, args, **kwargs):
+        self.separator = ''
+        self.joined = args
+        super(Concat, self).__init__(args, **kwargs)
+
+    def parse_args(self, args):
+        if not isinstance(args, list):
+            raise ValueError(
+                'Illegal arguments passed to {0} function. '
+                'Expected: [arg1\\function1, arg2\\function2, ...]'
+                'but got: {1}.'.format(self.name, args))
+
     def validate(self, plan):
-        pass
+        if plan.version.definitions_version < (1, 1):
+            raise exceptions.FunctionValidationError(
+                'Using {0} requires using dsl version 1_1 or '
+                'greater, but found: {1} in {2}.'
+                .format(self.name, plan.version, self.path))
 
     def evaluate(self, handler):
-        return handler.get_secret(self.secret_id)
+        for joined_value in self.joined:
```

Comment:
not clear naming, not sure the logic here

Reply:
this is concat, so joins values, so those are joined
anyway, this isn't changed in this PR, (used to be the same before), only git marks it as added+removed because concat was moved up

@davidginzbourg perhaps you can move it down so that git doesn't mark those lines as changed

anyway, this is fiiiine